### ISSUE TITLE
MapSequence has a non-functioning keyword `derotate`

### ISSUE THREAD

**Issue by ayshih** [sunpy/sunpy#3507]
The `MapSequence` constructor accepts the keyword `derotate`, with the following documentation:
```
derotate : `bool`
    Apply a derotation to the data. Default to False.
```

The code that is called when `derotate == True` is:
```python
def _derotate(self):
    """Derotates the layers in the MapSequence"""
    pass
```

That is, the keyword doesn't actually do anything.  I'm mystified.

**Comment by Raahul-Singh** [comment]
Hi, I was going through the docs and realized that a lot can be done here.
What would you recommend as the optimal funtionality that needs to be added here?

**Comment by nabobalis** [comment]
Probably needs to be removed. Adding that feature might be non trivial.

**Comment by nabobalis** [comment]
Having said that, I wonder if all you want to do is call rotate on each map.

**Comment by Raahul-Singh** [comment]
Just for the challenge, I'd try implementing the non trivial solution. I assume ``mapsequence_solar_derotate`` which uses  ``sunpy.physics.solar_rotation`` would work, or should I go for differential rotation?
Also, could you suggest some data with more than two maps so that I can test whether I am on the right track or not.

**Comment by nabobalis** [comment]
It would be what function or code that map.rotate uses. Probably to rotate the following maps to the first one in the sequence.  

In terms of data, you'll need to acquire some (using one of the gallery examples as a base) or construct one from the test data.

**Comment by Raahul-Singh** [comment]
Cool. Thanks! :smiley:

**Comment by wtbarnes** [comment]
`map.rotate` is used for rotating about an axis perpendicular to the plane of the image. My understanding is that the intention of the `derotate` kwarg in `MapSequence` was to align all maps in the sequence to some reference image. This can be done in various ways, but ultimately the rotation is going to be about the rotation axis of the Sun rather than about the axis perpendicular to the plane of the image so I don't think we want to be using `.rotate` here.

One approach would be to do this using the existing differential rotation functionality, though it would probably be better to wait until #3537 and #3540 are merged.

**Comment by nabobalis** [comment]
Ah I was mistaken. I just think `derotate` is a strange name for that since it's not the opposite of `rotate`.

**Comment by Raahul-Singh** [comment]
@wtbarnes Thanks! I'll pick this up after those PRs are merged.

**Comment by ayshih** [comment]
While implementing actual functionality would be great, in the meantime I suggest bugfixing existing releases such that using this keyword raises a `NotImplementedError`.

**Comment by Raahul-Singh** [comment]
@ayshih Would a simple ``raise NotImplementedError("This functionality has not yet been implemented.")`` in the function body suffice for now?

**Comment by ayshih** [comment]
Yup (plus a test)

**Comment by Raahul-Singh** [comment]
I've added a PR #3613 
This is my first time working with ``pytest``. Please let me know if any improvements can be made.

**Comment by wtbarnes** [comment]
Closed by #3613
